transaction/core/src/ring_signature/rct_bulletproofs.rs

```diff
@@ -248,68 +246,45 @@ fn sign_with_balance_check<CSPRNG: RngCore + CryptoRng>(
         }
     }
 
-    // Blindings for pseudo_outputs. All but the last are random.
-    // Constructing blindings in this way ensures that sum_of_outputs - sum_of_pseudo_outputs = 0
-    // if the sum of outputs and the sum of pseudo_outputs have equal value.
-    let mut pseudo_output_blindings: Vec<Scalar> = Vec::new();
-    for _i in 0..num_inputs - 1 {
-        pseudo_output_blindings.push(Scalar::random(rng));
-    }
-    // The implicit fee output is ommitted because its blinding is zero.
-    let sum_of_output_blindings: Scalar = output_values_and_blindings
-        .iter()
-        .map(|(_, blinding)| blinding)
-        .sum();
-
-    let sum_of_pseudo_output_blindings: Scalar = pseudo_output_blindings.iter().sum();
-    let last_blinding: Scalar = sum_of_output_blindings - sum_of_pseudo_output_blindings;
-    pseudo_output_blindings.push(last_blinding);
-
-    // Create Range proofs for outputs and pseudo-outputs.
-    let pseudo_output_values_and_blindings: Vec<(u64, Scalar)> = input_secrets
-        .iter()
-        .zip(pseudo_output_blindings.iter())
-        .map(|((_, value, _), blinding)| (*value, *blinding))
-        .collect();
-
-    let (range_proof, commitments) = {
-        let values_and_blindings: Vec<(u64, Scalar)> = pseudo_output_values_and_blindings
-            .iter()
-            .chain(output_values_and_blindings.iter())
-            .map(|(value, blinding)| (*value, *blinding))
-            .collect();
-
-        // The implicit fee output is omitted from the range proof because it is known.
-
-        let (values, blindings): (Vec<_>, Vec<_>) = values_and_blindings.into_iter().unzip();
-        generate_range_proofs(&values, &blindings, rng).map_err(|_e| Error::RangeProofError)?
-    };
+    let input_values: Vec<u64> = input_secrets.iter().map(|(_, value, _)| *value).collect();
+    let (output_values, output_blindings): (Vec<u64>, Vec<Scalar>) =
+        output_values_and_blindings.iter().cloned().unzip();
 
     if check_value_is_preserved {
```

Comment:
The point of this check while signing is to give clients immediate feedback that their transaction will be rejected during verification. Previously, we had copied part of signature verification. That's a really convoluted way to check if the inputs equal the outputs plus fee. The unit tests check that valid signatures pass verification, and that seems like the right way to cover this functionality.